pkg/deploy/deploy.go

```diff
@@ -85,8 +97,133 @@ func (o *deployHandler) ApplyOpenShift(openshift v1alpha2.Component) error {
+		ObjectMeta: metav1.ObjectMeta{
+			// TODO: Check if the name is K8s valid.
+			Name: o.componentName + "-" + o.appName + "-" + command.Id + "-" + dfutil.GenerateRandomString(3), // TODO: Is there a function to return the standard odo names?
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  containerComp.Name,
+							Image: containerComp.Image,
+							// TODO: Should we use the command and args already defined inside the container component?
+							// TODO: What if the 'sh' binary is not always available?
+							Command:         []string{"/bin/sh"},
+							Args:            getCmdline(command),
+							ImagePullPolicy: "IfNotPresent",
+						},
+					},
+					// Set the policy to `Never` so that it keeps the pod around and they can be used to debug.
+					RestartPolicy: "Never",
+				},
+			},
+			BackoffLimit:   pointer.Int32(2),
+			CompletionMode: &completionMode,
+			// we delete jobs before exiting this function but setting this as a backup in case DeleteJob fails
+			TTLSecondsAfterFinished: pointer.Int32(60),
+		},
+	}
+
+	// Set labels and annotations
+	job.SetLabels(odolabels.GetLabels(o.componentName, o.appName, component.GetComponentRuntimeFromDevfileMetadata(o.devfileObj.Data.GetMetadata()), odolabels.ComponentDeployMode, false))
+	job.Annotations = map[string]string{}
+	odolabels.AddCommonAnnotations(job.Annotations)
+	odolabels.SetProjectType(job.Annotations, component.GetComponentTypeFromDevfileMetadata(o.devfileObj.Data.GetMetadata()))
+
+	//	Make sure there are no existing jobs
+	checkAndDeleteExistingJob := func() {
+		items, dErr := o.kubeClient.ListJobs(odolabels.GetSelector(o.componentName, o.appName, odolabels.ComponentDeployMode, false))
+		if dErr != nil {
+			klog.V(4).Infof("failed to list jobs; cause: %s", dErr.Error())
+			return
+		}
+		partialJobName := o.componentName + "-" + o.appName + "-" + command.Id
+		for _, item := range items.Items {
+			if strings.Contains(item.Name, partialJobName) {
+				dErr = o.kubeClient.DeleteJob(item.Name)
+				if dErr != nil {
+					klog.V(4).Infof("failed to delete job %q; cause: %s", item.Name, dErr.Error())
+				}
+			}
+		}
+	}
+	checkAndDeleteExistingJob()
+
+	log.Sectionf("Executing command in container (command: %s)", command.Id)
+	spinner := log.Spinnerf("Executing %q", command.Exec.CommandLine)
+	defer spinner.End(false)
+
+	var createdJob *batchv1.Job
+	createdJob, err = o.kubeClient.CreateJob(job, "")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = o.kubeClient.DeleteJob(createdJob.Name)
+		if err != nil {
+			klog.V(4).Infof("failed to delete job %q; cause: %s", createdJob.Name, err)
+		}
+	}()
+
+	// Print the tip to use `odo logs` if the command is still running after 1 minute
+	go func() {
+		time.Sleep(1 * time.Minute)
+		log.Info("\nTip: Run `odo logs --deploy --follow` to get the logs of the command output.")
```

Comment:
https://go.dev/play/p/CBxlxFxDZGM
Add a watch on a channel to know if the job has executed. Otherwise the tip would be printed even if the job in question has completed, but the odo process is still running.